model/remote_cluster.go

```diff
@@ -35,48 +41,219 @@ func (rc *RemoteCluster) PreSave() {
 	if rc.CreateAt == 0 {
 		rc.CreateAt = GetMillis()
 	}
+	rc.fixTopics()
+}
+
+func (rc *RemoteCluster) IsValid() *AppError {
+	if !IsValidId(rc.RemoteId) {
+		return NewAppError("RemoteCluster.IsValid", "model.cluster.is_valid.id.app_error", nil, "id="+rc.RemoteId, http.StatusBadRequest)
+	}
+
+	if rc.DisplayName == "" {
+		return NewAppError("RemoteCluster.IsValid", "model.cluster.is_valid.name.app_error", nil, "display_name empty", http.StatusBadRequest)
+	}
 
-	if rc.LastPingAt == 0 {
-		rc.LastPingAt = rc.CreateAt
+	if rc.CreateAt == 0 {
+		return NewAppError("RemoteCluster.IsValid", "model.cluster.is_valid.create_at.app_error", nil, "create_at=0", http.StatusBadRequest)
 	}
+	return nil
 }
 
-func (rc *RemoteCluster) IsValid() *AppError {
-	if !IsValidId(rc.Id) {
-		return NewAppError("RemoteCluster.IsValid", "model.cluster.is_valid.id.app_error", nil, "id="+rc.Id, http.StatusBadRequest)
+func (rc *RemoteCluster) PreUpdate() {
+	rc.fixTopics()
+}
+
+// fixTopics ensures all topics are separated by one, and only one, space.
+func (rc *RemoteCluster) fixTopics() {
+	trimmed := strings.TrimSpace(rc.Topics)
+	if trimmed == "" {
+		rc.Topics = ""
+		return
+	}
+	if trimmed == "*" {
+		rc.Topics = "*"
+		return
 	}
 
-	if rc.ClusterName == "" {
-		return NewAppError("RemoteCluster.IsValid", "model.cluster.is_valid.name.app_error", nil, "cluster_name empty", http.StatusBadRequest)
+	var sb strings.Builder
+	sb.WriteString(" ")
+
+	ss := strings.Split(rc.Topics, " ")
```

Comment:
Just pointing out that this could cause issues if the topics have other whitespace characters like tabs in between. Not sure how robust we want to be on that front.

Reply:
Agreed.  Topics are not user input and must be coded for in the app.  They won't have tabs or other whitespace within the topic.